nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -791,8 +791,14 @@ public TransferResult<Nullness, NullnessStore> visitArrayAccess(
     Nullness resultNullness;
     // Unsoundly assume @NonNull, except in JSpecify mode where we check the type
     if (config.isJSpecifyMode()) {
-      Symbol arraySymbol = ASTHelpers.getSymbol(node.getArray().getTree());
+      Symbol arraySymbol;
       boolean isElementNullable = false;
+      ExpressionTree arrayExpr = node.getArrayExpression();
```

Comment:
add a comment indicating what is going on here (we special-case for-each loops)

Reply:
Fixed